pkg/apis/elasticsearch/v1/validations.go

```diff
@@ -19,18 +20,19 @@ import (
 
 const (
 	cfgInvalidMsg            = "Configuration invalid"
+	duplicateNodeSets        = "NodeSet names must be unique"
+	invalidNamesErrMsg       = "Elasticsearch configuration would generate resources with invalid names"
+	invalidSanIPErrMsg       = "Invalid SAN IP address. Must be a valid IPv4 address"
 	masterRequiredMsg        = "Elasticsearch needs to have at least one master node"
-	parseVersionErrMsg       = "Cannot parse Elasticsearch version. String format must be {major}.{minor}.{patch}[-{label}]"
+	mixedRoleConfigMsg       = "Detected a combination of node.roles and %s. Use only node.roles"
```

Comment:
Because there's no good term to describe `node.{role}` config settings, I chose to just list them so that the user understands what needs to be removed. Do you think it's not necessary?

Reply:
Ah, that's fine. I was referring to "Use only node.roles", I thought it might be better to just say that either one approach or the other should be used instead of pointing users to use `node.roles` specifically.

Reply:
If it is preferred then I'm fine leaving it as it is.